datafusion/core/src/datasource/schema_adapter.rs

```diff
@@ -167,55 +186,95 @@ impl SchemaAdapter for DefaultSchemaAdapter {
 
 /// The SchemaMapping struct holds a mapping from the file schema to the table schema
 /// and any necessary type conversions that need to be applied.
+///
+/// This needs both the projected table schema and full table schema because its different
+/// functions have different needs. The `map_batch` function is only used by the ParquetOpener to
+/// produce a RecordBatch which has the projected schema, since that's the schema which is supposed
+/// to come out of the execution of this query. `map_partial_batch`, however, is used to
```

Comment:
This comment appears to stop mid sentence

Reply:
Thanks for catching that, just fixed :)